pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -351,7 +357,8 @@ class Timestamp(_Timestamp):
         second=None,
         microsecond=None,
         nanosecond=None,
-        tzinfo=None
+        tzinfo=None,
+        fold=None
```

Comment:
Think it's okay to default `fold=0` like `datetime.datetime` does.

Reply:
Setting fold to `None` by default allows us to use the fold attribute from ts_input when nothing was supplied by the user:
```
        if getattr(ts_input, 'fold', None) is not None and fold is None:
            fold = ts_input.fold
```

As a result, we get to pass fold through when constructing from pydatetime (and adjust value for that fold) or override fold if it's explicitly supplied by the user:
```
Timestamp(ts_input=datetime(2019, 10, 27, 1, 30, 0, 0, fold=0)) => ts.fold == 0, ts.value == 1572136200000000000)

Timestamp(ts_input=datetime(2019, 10, 27, 1, 30, 0, 0, fold=1)) => ts.fold == 1, ts.value == 1572139800000000000)

Timestamp(ts_input=datetime(2019, 10, 27, 1, 30, 0, 0, fold=1), fold=0) => ts.fold == 0, ts.value == 1572136200000000000)

Timestamp(ts_input=datetime(2019, 10, 27, 1, 30, 0, 0, fold=0), fold=1) => ts.fold == 1, ts.value == 1572139800000000000)
```
Without the default `fold=None`, it's hard to establish priority between `ts_input.fold` and supplied `fold`.
Another way to do this might be possible, but I would like for us to support this behavior.

Reply:
See my comment on raising an error on conflict between `ts_input.fold` and `fold`.